admin/src/Models/Member.js

```diff
@@ -1,12 +1,22 @@
 import Base from './Base';
+import { showError } from "../message";
 
 
 export default class Member extends Base {
-    
+
     deleteConfirmMessage() {
         return `Are you sure you want to delete member ${this.firstname} ${this.lastname}?`;
     }
-    
+
+    validateEmail() {
+        if (this.email.includes("@") && this.email.includes(".")) {
+            return true
+        } else {
+            showError("Felaktigt format på mailadressen");
+            return false
+        }
+    }
+
     canSave() {
         return this.isDirty() && this.email.length > 0 && this.firstname.length > 0;
```

Comment:
I implemented it that way first, but the problem was that canSave() is executed every time a field in the form is modified by the user. This led to a lot of notifications about incorrect email format appearing. Therefore, I moved validateEmail() outside canSave() to be able to keep the notifications to explain why the user can't save the form when any of the email criterias are missing.

Reply:
I implemented it that way first, but the problem was that canSave() is executed every time a field in the form is modified by the user. This led to a lot of notifications about incorrect email format appearing. Therefore, I moved validateEmail() outside canSave() to be able to keep the notifications to explain why the user can't save the form when any of the email criterias are missing.

Reply:
I think you should "lift out" the error message logic from this method. It makes more sense that the caller creates the popup depending on context. That also conforms better with the method name "validateEmail".

If you save or return the reason why the member cannot be saved, then you can use that in the UI, like making the save button greyed out and with a tool tip that says `email validation failed. Missing @`.

Reply:
That sounds reasonable! We realised that it might be better to check more things than just the email if tooltips should be used. Maybe return a string from canSave. But that could be applied to quite many canSave functions, therefore we thought that we can leave it for now. So I just changed to checking for `@` and `.` in canSave , without generating an error message or similar.